quint/mempool/Mempool.qnt

```diff
@@ -0,0 +1,191 @@
+// -*- mode: Bluespec; -*-
+// A mempool is a replicated set of transactions which is used as an input by a ledger.
+// Below, we follow the specification given here:
+// https://github.com/cometbft/knowledge-base/blob/main/protocols/mempool-overview.md
+
+module ABCI {
+
+    import System.* from "System"
+    import Ledger.* from "Ledger"
+
+    // a transaction is committed at most once
+    pure def isValid(l: LedgerState, p: Process, t: Tx): bool = {
+      not(isCommittedFor(l, p, t))
+    }
+
+}
+
+module Mempool {
+
+    import Spells.* from "Spells"
+    import System.* from "System"
+    import Ledger.* from "Ledger"
+    import ABCI.*
+
+    type MempoolState = {
+      pool: Process -> Set[Tx],
```

Comment:
Don't we actually have a FIFO (in production)?

Reply:
True. Again, this is a simplification because decision in which the same transaction is decided over and over are stuttering from a progress perspective.